include/zcbor_diag_print.h

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZCBOR_DIAG_PRINT_H__
+#define ZCBOR_DIAG_PRINT_H__
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef ZCBOR_DIAG_PRINT_FUNC
+#include <zephyr/sys/printk.h>
+#define zcbor_do_diag_print(...) printk(__VA_ARGS__)
+#else
+#define zcbor_do_diag_print(...) ZCBOR_DIAG_PRINT_FUNC(__VA_ARGS__)
+#endif
+
+static bool indent_printed = false;
+
+__attribute__((used))
+static void zcbor_print_indent(uint8_t indent_len)
+{
+	if (!indent_printed) {
+		for (int i = 0; i < indent_len; i++) {
+			zcbor_do_diag_print("| ");
+		}
+		indent_printed = true;
+	}
+}
+
+__attribute__((used))
+static void zcbor_print_newline(void)
+{
+	zcbor_do_diag_print("\r\n");
+	indent_printed = false;
+}
+
+#define STR_COL_WIDTH 16
```

Comment:
The width of a string column? Or the width of a row? Could we make the variable name more verbose?